src/txn/txn_rollback_to_stable.c

```diff
@@ -281,50 +281,30 @@ __rollback_row_modify(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW *rip, WT_U
 }
 
 /*
- * __rollback_check_if_txnid_non_committed --
- *     Check if the transaction id is non committed.
+ * __rollback_txn_visible_id --
+ *     Check if the transaction id is visible or not.
  */
 static bool
-__rollback_check_if_txnid_non_committed(WT_SESSION_IMPL *session, uint64_t txnid)
+__rollback_txn_visible_id(WT_SESSION_IMPL *session, uint64_t id)
 {
     WT_CONNECTION_IMPL *conn;
-    bool found;
 
     conn = S2C(session);
 
-    /* If not recovery then assume all the data as committed. */
+    /* If not recovery then assume all the data as visible. */
     if (!F_ISSET(conn, WT_CONN_RECOVERING))
-        return (false);
+        return (true);
 
     /*
      * Only full checkpoint writes the metadata with snapshot. If the recovered checkpoint snapshot
-     * details are zero then return false i.e, updates are committed.
-     */
-    if (conn->recovery_ckpt_snap_min == 0 && conn->recovery_ckpt_snap_max == 0)
-        return (false);
-
-    /*
-     * Snapshot data:
-     *	ids < recovery_ckpt_snap_min are committed,
-     *	ids > recovery_ckpt_snap_max are non committed,
-     *	everything else is committed unless it is found in the recovery_ckpt_snapshot array.
+     * details are none then return false i.e, updates are visible.
```

Comment:
I am curious if having snap min and max set to `WT_TXN_NONE` means we do not have snapshot for recovery?

Reply:
Looks like it @kommiharibabu

Reply:
Maybe we can rewrite this comment as : If the recovered checkpoint snapshot's min and max are set to `WT_TXN_NONE`, we can infer that we do not have snapshot for recovery and in that case all updates should be visible.